src/main/java/com/openlattice/datastore/data/controllers/DataController.java

```diff
@@ -670,6 +670,11 @@ public Integer deleteEntities(
             @PathVariable( ENTITY_SET_ID ) UUID entitySetId,
             @RequestBody Set<UUID> entityKeyIds,
             @RequestParam( value = TYPE ) DeleteType deleteType ) {
+
```

Comment:
unrelated but just noticed we're not calling `ensureWriteAccess` here which we should definitely be doing... (also that we call `ensureReadAccess` if it's a soft delete, which we should not be doing)

Reply:
I thought that `ensureReadAccess` looked shifty when I first read this

Reply:
LATTICE-1924

Reply:
cool thanks!

Reply:
This is a pretty old debate actually, but last time I asked Matthew for permission checks on soft delete, the answer was this:
"Julia [15:55]
in soft deletes, which kind of permissions are required? on entityset and on properties?
because here I’m seeing, that for entityset it’s only checking read
and properties only write

matthew [16:12]
That's correct
soft delete is a lot more permissive
because it doesn't remove data"

Reply:
yeah +1 since we're checking property type permissions and permission inheritance isn't a thing -- although why is read needed on the entity set?

Reply:
I'm not sure.. maybe because this is basically a write to the entity set?

Reply:
You need read on entity set metadata to issue any mutations against an entity set that rely upon that data (which is pretty much anything except maybe reading a specific property).

Reply:
So, if the pattern is first checking permissions, it shouldn't be allowed to clean only authorized properties? Right now it's working like that